### PR TITLE
media-libs/opencollada: Version bump to 1.6.68

### PR DESCRIPTION
**Pull request by redchillipadi**
Version bump to 1.6.68 as required by blender-2.82a.
The ebuild has been updated to EAPI 7.

The null conversion QA error was fixed by initialising std::vector with
zero. I resolved the insecure DAEValidator RUNPATH QA error by clearing
the path with app-admin/chrpath and installing its library
libDAEValidatorLibrary.so

Signed-off-by: Adrian Grigo <agrigo2001@yahoo.com.au>
Closes: https://bugs.gentoo.org/721682
Package-Manager: Portage-2.3.99, Repoman-2.3.22

### CODE REVIEW

File: media-libs/opencollada/opencollada-1.6.68.ebuild [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils eutils flag-o-matic
```
Comment (juippis) [inline media-libs/opencollada/opencollada-1.6.68.ebuild:6]: *cmake-utils.eclass* is being deprecated, use *cmake.eclass* instead.

File: media-libs/opencollada/opencollada-1.6.68.ebuild [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+HOMEPAGE="http://www.opencollada.org/"
+SRC_URI="https://github.com/KhronosGroup/OpenCOLLADA/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+IUSE="static-libs"
+
+RDEPEND="dev-libs/libpcre
+	dev-libs/libxml2
```
Comment (juippis) [inline media-libs/opencollada/opencollada-1.6.68.ebuild:18]: Add slot operators.

File: media-libs/opencollada/opencollada-1.6.68.ebuild [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	dev-libs/zziplib
+	media-libs/lib3ds
+	sys-libs/zlib
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+	app-admin/chrpath
```
Comment (juippis) [inline media-libs/opencollada/opencollada-1.6.68.ebuild:26]: Please sort while at it.

File: media-libs/opencollada/opencollada-1.6.68.ebuild [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	)
+
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	echo "LDPATH=/usr/$(get_libdir)/opencollada" > "${T}"/99${PN} || die "echo failed"
+	doenvd "${T}"/99${PN}
```
Comment (juippis) [inline media-libs/opencollada/opencollada-1.6.68.ebuild:67]: Can be done with a simple `newenvd`. See https://gitweb.gentoo.org/repo/gentoo.git/tree/mail-client/thunderbird-bin/thunderbird-bin-68.8.0.ebuild#n151 for example.

File: media-libs/opencollada/opencollada-1.6.68.ebuild [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+	# Remove insecure DAEValidator RUNPATH and install DAEValidator library
+	dolib.so "${BUILD_DIR}/lib/libDAEValidatorLibrary.so"
+	/usr/bin/chrpath -d "${BUILD_DIR}/bin/DAEValidator"
```
Comment (juippis) [inline media-libs/opencollada/opencollada-1.6.68.ebuild:71]: Why not just call `chrpath`? Would work in prefix systems. Also append with `|| die`.

File: media-libs/opencollada/opencollada-1.6.68.ebuild [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+	dobin "${BUILD_DIR}/bin/DAEValidator"
+	dobin "${BUILD_DIR}/bin/OpenCOLLADAValidator"
+	# Need to be in same directory as above binaries
+	docinto "/usr/bin"
```
Comment (juippis) [inline media-libs/opencollada/opencollada-1.6.68.ebuild:76]: Is there no upstream issue/patch about this...?

File: media-libs/opencollada/opencollada-1.6.68.ebuild (hunk 2 above)
Comment (a17r) [reply media-libs/opencollada/opencollada-1.6.68.ebuild:18]: only libxml2 has a slot other than "0", so you can make it depend on `:2` indeed, but no operator needed.

**Pull request closed without merging**